pkg/component/component.go

```diff
@@ -326,6 +325,31 @@ func (r *RuntimeSpecs) PolicyToComponents(policy map[string]interface{}) ([]Comp
 	return components, componentIdsInputMap, nil
 }
 
+// Injects or creates a policy.revision sub-object in the input map.
+func injectInputPolicyID(fleetPolicy map[string]interface{}, input map[string]interface{}) {
+	// If there is no top level fleet policy revision, there's nothing to inject.
+	revision, exists := fleetPolicy["revision"]
+	if !exists || input == nil {
```

Comment:
nit: if checking input for nil then might as well do it before the map lookup and return